Extensions/timestamps.js

```diff
@@ -108,37 +107,58 @@ XKit.extensions.timestamps = new Object({
 
 		if (XKit.interface.where().search) {
 			this.in_search = true;
-			XKit.tools.add_css(`
-				.xtimestamp-in-search {
-					position: absolute;
-					top: 32px;
-					color: rgb(168,177,184);
-					font-size: 10px;
-				}`, "timestamps");
 		}
 
 		if (this.preferences.format.value === "") {
 			this.preferences.format.value = "MMMM Do YYYY, h:mm:ss a";
 		}
 
 		this.check_quota();
+
+		$(document).on("click", ".xkit-timestamp-failed-why", function() {
+			XKit.window.show("Timestamp loading failed.", "This might be caused by several reasons, such as the post being removed, becoming private, or the Tumblr server having a problem that it can't return the page required by XKit to load you the timestamp.", "error", "<div id=\"xkit-close-message\" class=\"xkit-button\">OK</div></div>");
+		});
+
+		if (XKit.page.react) {
+			XKit.css_map.getCssMap()
+			.then(() => {
+				this.posts_class = XKit.css_map.keyToCss("post");
+				this.reblogs_class = XKit.css_map.keyToCss("reblog");
+				this.reblog_headers_class = XKit.css_map.keyToCss("reblogHeader");
+				this.blog_link_class = XKit.css_map.keyToCss("blogLink");
+			});
+
+			if (this.preferences.posts.value || (this.preferences.inbox.value && XKit.interface.where().inbox)) {
+				this.react_add_timestamps();
+				XKit.post_listener.add("timestamps", this.react_add_timestamps);
+			}
+
+			if (this.preferences.reblogs.value !== "off") {
+				this.react_add_reblog_timestamps();
+				XKit.post_listener.add("timestamps", this.react_add_reblog_timestamps);
+			}
+
+			if (this.preferences.only_on_hover.value) {
+				XKit.tools.add_css(`.xtimestamp { display: none; } ${this.posts_class.split(", ").map(x => x + ":hover .xtimestamp").join(", ")} { display: block; }`, "timestamps_on_hover");
```

Comment:
`this.posts_class` isn't guaranteed to be defined here. 

`XKit.css_map.getCssMap()` is an asynchronous function, which means that technically, it (and any functions attached with `.then()`) may not have finished executing by the time the next line, or any of the rest of the code block executes.

all the code in the `if (XKit.page.react)` block (currently below the `.then()` function) should be in the `.then()` function to guarantee that the necessary values have been set.

Reply:
That's a great point, forgot about that